python/tests/unit/arcticdb/version_store/test_sort_merge.py

```diff
@@ -222,50 +217,303 @@ def test_prune_previous(lmdb_library):
+            lib.sort_and_finalize_staged_data("sym", mode=StagedDataFinalizeMethod.APPEND)
+        assert "col_1" in str(exception_info.value)
+        assert "col_2" in str(exception_info.value)
+
+    def test_cannot_append_reordered_columns(self, lmdb_library):
+        lib = lmdb_library
+        self.init_symbol(lib, "sym")
+        
+        df = pd.DataFrame({"col_1": [1.4], "col_0": [5], "col_2": ["val"]}, index=pd.DatetimeIndex([pd.Timestamp(2024, 1, 1)]))
+        lib.write("sym", df, staged=True)
+        with pytest.raises(SchemaException) as exception_info:
+            lib.sort_and_finalize_staged_data("sym", mode=StagedDataFinalizeMethod.APPEND)
+        assert "col_0" in str(exception_info.value)
+        assert "col_1" in str(exception_info.value)
+        assert "col_2" in str(exception_info.value)
+
+    def test_cannot_promote_type(self, lmdb_library):
+        lib = lmdb_library
+        self.init_symbol(lib, "sym")
+        
+        df = pd.DataFrame({"col_0": np.array([1], dtype="int64"), "col_1": [5], "col_2": ["val"]}, index=pd.DatetimeIndex([pd.Timestamp(2024, 1, 1)]))
+        lib.write("sym", df, staged=True)
+        with pytest.raises(SchemaException) as exception_info:
+            lib.sort_and_finalize_staged_data("sym", mode=StagedDataFinalizeMethod.APPEND)
+        assert "col_0" in str(exception_info.value)
+        assert "INT32" in str(exception_info.value)
+        assert "INT64" in str(exception_info.value)
+
+        df = pd.DataFrame({"col_0": np.array([1], dtype="int16"), "col_1": [5], "col_2": ["val"]}, index=pd.DatetimeIndex([pd.Timestamp(2024, 1, 1)]))
+        lib.write("sym", df, staged=True)
+        with pytest.raises(SchemaException) as exception_info:
+            lib.sort_and_finalize_staged_data("sym", mode=StagedDataFinalizeMethod.APPEND)
+        assert "col_0" in str(exception_info.value)
+        assert "INT32" in str(exception_info.value)
+        assert "INT16" in str(exception_info.value)
+
+    def test_cannot_add_new_columns(self, lmdb_library):
+        lib = lmdb_library
+        self.init_symbol(lib, "sym")
+        
+        df = pd.DataFrame({"col_0": np.array([1], dtype="int32"), "col_1": [5], "col_2": ["val"], "col_3": [1]}, index=pd.DatetimeIndex([pd.Timestamp(2024, 1, 1)]))
+        lib.write("sym", df, staged=True)
+        with pytest.raises(SchemaException) as exception_info:
+            lib.sort_and_finalize_staged_data("sym", mode=StagedDataFinalizeMethod.APPEND)
+        assert "col_3" in str(exception_info.value)
+        
+        
+# This was a added as a bug repro for GH issue #1795.
+def test_two_columns_with_different_dtypes(lmdb_library_dynamic_schema):
+    lib = lmdb_library_dynamic_schema
+
+    idx1 = pd.DatetimeIndex([
+        pd.Timestamp("2024-01-02")
+    ])
+    df1 = pd.DataFrame({
+         "a": np.array([1], dtype="float"),
+         "b": np.array([22250], dtype="int64")
+    }, index=idx1)
+    
+    b = np.array([-53979, -53973], dtype="int64")
```

Comment:
I assume the weirdly specific values `1/22250/-53979/-53973` aren't necessary to repro the original bug? Replace them with something more standard, otherwise it seems like these values are special somehow